query/engine.go

```diff
@@ -9,71 +9,91 @@ import (
 	"github.com/polarsignals/arcticdb/query/physicalplan"
 )
 
-type Engine struct {
+type Engine interface {
+	ScanTable(name string) QueryBuilder
+	ScanSchema(name string) QueryBuilder
+}
```

Comment:
Better to move this to where it's actually called.

Reply:
agreed. done.